src/rdb_protocol/terms/sort.cc

```diff
@@ -83,9 +83,9 @@ class orderby_term_t : public op_term_t {
         /* Add a sorting to the table if we're doing indexed sorting. */
         } else if (index.has()) {
             rcheck(tbl_slice.has(), base_exc_t::LOGIC,
-                   "Indexed order_by can only be performed on a TABLE or TABLE_SLICE.");
+                   "Indexed order_by can only be performed on a TABLE or TABLE_SLICE, not on a `NAME_OF_TYPE_ORDER_BY_WAS_CALLED_ON`.  Make sure ORDER_BY comes before any transformations such as MAP or FILTER.");
```

Comment:
@mlucy Hmm it looks like we are somewhat inconsistent on the capitalization of command names. I thought we used lower-case for commands, and upper-case for type names. But it looks like we're using both variants. (e.g. `ORDER_BY` in https://github.com/rethinkdb/rethinkdb/blob/next/src/rdb_protocol/term_walker.cc#L103 but `order_by` in this message or in https://github.com/rethinkdb/rethinkdb/blob/next/src/rdb_protocol/order_util.cc#L24).

I personally think the lower-case version looks nicer, but I don't particularly care either way. We should probably pick one format though and use it for future changes. Any opinions on which one it should be?